### PR TITLE
fix(processor): support distributed subquery in new cluster framework

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

fix(processor): support distributed subquery in new cluster framework and add new `EXPLAIN FRAGMENTS` syntax:
```
EXPLAIN FRAGMENTS select COUNT() from numbers(10) GROUP BY number % 3;
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| explain                                                                                                                                                                     |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Fragment 0:                                                                                                                                                                 |
|   DataExchange: Shuffle                                                                                                                                                     |
|   AggregatorPartial: groupBy=[[(number % 3)]], aggr=[[COUNT()]]                                                                                                             |
|     Expression: (number % 3):UInt8 (Before GroupBy)                                                                                                                         |
|       ReadDataSource: scan schema: [number:UInt64], statistics: [read_rows: 10, read_bytes: 80, partitions_scanned: 1, partitions_total: 1], push_downs: [projections: [0]] |
|                                                                                                                                                                             |
| Fragment 2:                                                                                                                                                                 |
|   DataExchange: Merge                                                                                                                                                       |
|   Projection: COUNT():UInt64                                                                                                                                                |
|     AggregatorFinal: groupBy=[[(number % 3)]], aggr=[[COUNT()]]                                                                                                             |
|       Remote[receive fragment: 0]                                                                                                                                           |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

Fixes #issue

### CODE REVIEW

**Comment by BohuTANG** [comment]
6666...

**Comment by zhang2014** [comment]
Add new syntax:
```
mysql> EXPLAIN FRAGMENTS select COUNT() from numbers(10) GROUP BY number % 3;
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| explain                                                                                                                                                                     |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Fragment 0:                                                                                                                                                                 |
|   DataExchange: Shuffle                                                                                                                                                     |
|   AggregatorPartial: groupBy=[[(number % 3)]], aggr=[[COUNT()]]                                                                                                             |
|     Expression: (number % 3):UInt8 (Before GroupBy)                                                                                                                         |
|       ReadDataSource: scan schema: [number:UInt64], statistics: [read_rows: 10, read_bytes: 80, partitions_scanned: 1, partitions_total: 1], push_downs: [projections: [0]] |
|                                                                                                                                                                             |
| Fragment 2:                                                                                                                                                                 |
|   DataExchange: Merge                                                                                                                                                       |
|   Projection: COUNT():UInt64                                                                                                                                                |
|     AggregatorFinal: groupBy=[[(number % 3)]], aggr=[[COUNT()]]                                                                                                             |
|       Remote[receive fragment: 0]                                                                                                                                           |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
11 rows in set (0.02 sec)
Read 0 rows, 0.00 B in 0.003 sec., 0 rows/sec., 0.00 B/sec.
```

File: query/src/pipelines/processors/sinks/subquery_receive_sink.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        for column_index in 0..data_block.num_columns() {
+            let column = data_block.column(column_index);
+            let mut values = column.to_values();
+            self.input_columns[column_index].append(&mut values)
+        }
+
+        Ok(())
+    }
+}
+// #[async_trait::async_trait]
```
Comment (BohuTANG) [inline query/src/pipelines/processors/sinks/subquery_receive_sink.rs:83]: Theses codes not used anymore?

**Review by BohuTANG — APPROVED** [review]
lgtm

**Review by leiysky — APPROVED** [review]

**Comment by BohuTANG** [comment]
This PR add a new syntax for explain:
EXPLAIN FRAGMENTS
cc @soyeric128

**Comment by Xuanwo** [comment]
Just noticed that this PR is 6666

**Pull request merged**